src/Services/DataService.test.ts

```diff
@@ -619,13 +640,59 @@ describe("DataService", () => {
       const expectedResult = { fullText: "puppy" };
       const createRecordAction = fakeService.actions.createRecord(expectedResult, { onSuccess });
       const pushRecordAction = stub(fakeService.actions, "pushRecord");
-
+      
       mockAdapter.createItem.returns(Observable.of(expectedResult));
 
       fakeService.createRecordEpic(ActionsObservable.of(createRecordAction), store)
         .subscribe(noop, noop,
           () => {
-            assert.isTrue(mockAdapter.createItem.calledWithMatch(JSON.stringify(expectedResult)));
+            expect(mockAdapter.createItem.calledWithMatch(JSON.stringify(expectedResult))).to.be.true;
+          });
+    });
+
+    it("createRecordEpic should call normalize after deserialize", () => {
+      const onSuccess = spy();
+      const expectedResult = { fullText: "zella puppy" };
+      const createRecordAction = fakeService.actions.createRecord(expectedResult, { onSuccess });
+      const normalizedStub = stub(fakeService.mapper, "normalize");
+
+      stub(fakeService.serializer, "deserialize").returns(expectedResult);
+
+      fakeService.createRecordEpic(ActionsObservable.of(createRecordAction), store)
+        .subscribe(noop, noop,
+          () => {
+            expect(normalizedStub.firstCall.args[0]).to.equal(expectedResult);
+          });
+    });
+
+    it("createRecordEpic should call transform before serialize", () => {
```

Comment:
The name of this test and the one below should be switched.  The one is checking that we are calling `serialize` with the result of the `transform` and vice versa.